libraries/core_libs/network/src/tarcap/taraxa_capability.cpp

```diff
@@ -321,13 +321,38 @@ void TaraxaCapability::interpretCapabilityPacket(std::weak_ptr<dev::p2p::Session
       return;
     }
   }
-
+  SubprotocolPacketType packet_type = static_cast<SubprotocolPacketType>(_id);
+  if (syncing_state_->is_deep_pbft_syncing() && filterSyncIrrelevantPackets(packet_type)) {
+    LOG(log_dg_) << "Ignored " << convertPacketTypeToString(packet_type) << " because we are still syncing";
+    return;
+  }
   // TODO: we are making a copy here for each packet bytes(toBytes()), which is pretty significant. Check why RLP does
   //       not support move semantics so we can take advantage of it...
-  SubprotocolPacketType packet_type = static_cast<SubprotocolPacketType>(_id);
   thread_pool_->push(PacketData(packet_type, std::move(node_id), _r.data().toBytes()));
 }
 
+inline bool TaraxaCapability::filterSyncIrrelevantPackets(SubprotocolPacketType packet_type) {
+  switch (packet_type) {
+    case DagBlockPacket:
+    case GetDagSyncPacket:
+    case DagSyncPacket:
+    case TransactionPacket:
+    case VotePacket:
+    case GetVotesSyncPacket:
+    case VotesSyncPacket:
+    case PbftBlockPacket:
```

Comment:
all these case are basically transformed into the `if else` conditions that we dont need to check at all. I would call this method:

```
TaraxaCapability::isPbftSyncingRelevatPacket(SubprotocolPacketType packet_type) {
  switch (packet_type) {
     case TestPacket:
     case StatusPacket:
     case GetPbftSyncPacket:
     case PbftSyncPacket:
     case SyncedPacket:
       return true;
  }

  return false
}
```

Reply:
but the we will not catch bug if someone implements new type and do not set if it should be filtered out or not. And also compiler will do something like that

Reply:
also it is less clear what packet we are filtering or not

Reply:
quite the opposite I think. If I create some new packet type which is related to deep syncing and wont add it to this switch, I am pretty sure that some syncing tests would fail so this is covered and if that new type is not related to deep syncing I should not be forced to bother with this function at all because it should check only few selected types that are related to deep syncing, nothing else... And the other argument that compiler will do something like that, well I dont see how it actually could remove valid if else condition. This function will be called basically with all the possible packet types so no way compiler will optimize it. Logically we want to check only those few selected types, nothing else so we can get rid of 8 useless if else conditions + 8 lines of code

Reply:
yeah, you are probably right